src/app/pages/FastBtcPage/components/Deposit/SignatureValidation/index.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { isValidElement, useContext } from 'react';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { translations } from 'locales/i18n';
+import { DepositDetails } from '../DepositDetails';
+import { AddressQrCode, URIType } from 'app/components/Form/AddressQrCode';
+import { DepositContext } from '../../../contexts/deposit-context';
+import { validateSignatures } from '../../../../../../utils/signature/signatureValidation';
+import { useAccount } from 'app/hooks/useAccount';
+import { WalletContext } from '@sovryn/react-wallet';
+import { getContract } from '../../../../../../utils/blockchain/contract-helpers';
+import { FastBtcButton } from '../../FastBtcButton';
+import { StatusComponent } from 'app/components/Dialogs/TxDialog';
+import { TxStatus } from 'store/global/transactions-store/types';
+import styles from './index.module.scss';
+
+interface ISignatureValidationProps {
+  onClick: () => void;
+}
+
+export const SignatureValidation: React.FC<ISignatureValidationProps> = ({
+  onClick,
+}) => {
+  const { address, signatures } = useContext(DepositContext);
+  const { t } = useTranslation();
+  const account = useAccount();
+  const { chainId } = useContext(WalletContext);
+  const multisigAddress = getContract('fastBtcMultisig').address;
+  const [isSignatureValid, setSignatureValid] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [pageStatus, setPageStatus] = useState(TxStatus.PENDING);
+  const [statusText, setStatusText] = useState(
+    t(translations.fastBtcPage.deposit.validationScreen.validationTextLoading),
+  );
+
+  validateSignatures(
+    signatures,
+    address,
+    account,
+    chainId,
+    multisigAddress,
+  ).then(valid => {
+    setSignatureValid(valid);
+    setLoading(false);
+    valid ? setPageStatus(TxStatus.CONFIRMED) : setPageStatus(TxStatus.FAILED);
+    valid
+      ? setStatusText(
+          t(
+            translations.fastBtcPage.deposit.validationScreen
+              .validationTextSuccess,
+          ),
+        )
+      : setStatusText(
+          t(
+            translations.fastBtcPage.deposit.validationScreen
+              .validationTextError,
+          ),
+        );
+  });
```

Comment:
Can probably refactor this part to work same as change on line above (move condition inside of a single `setStatusText()` call) - see which one looks better style-wise